tests/unit/test_data_interface.py

```diff
@@ -0,0 +1,277 @@
+
+    def _testSearchMethodInContainer(
+            self, containerAccessorMethod, clientSearchMethod,
+            containerIteratorMethod):
+        for container in containerIteratorMethod:
+            containerId = container.getId()
+            repoObjs = getattr(container, containerAccessorMethod)()
+            numRepoObjs = len(repoObjs)
+            if numRepoObjs == 0:
+                continue
+            pageSizes = self._getPageSizes(numRepoObjs)
+            for pageSize in pageSizes:
+                self._client.setPageSize(pageSize)
+                clientObjs = list(clientSearchMethod(containerId))
+                for repoObj, clientObj in utils.zipLists(
+                        repoObjs, clientObjs):
+                    self._assertEqual(repoObj, clientObj)
+
+    def setUp(self):
+        self._repo = datarepo.SqlDataRepository(paths.testDataRepo)
+        self._repo.open(datarepo.MODE_READ)
+        self._backend = backend.Backend(self._repo)
+        self._client = client.LocalClient(self._backend)
+
+    def testGetBioSample(self):
+        self._testGetMethod(
+            self._repo.allBioSamples, self._client.getBioSample)
+
+    def testGetIndividual(self):
+        self._testGetMethod(
+            self._repo.allIndividuals, self._client.getIndividual)
+
+    def testGetDataset(self):
+        self._testGetMethod(
+            self._repo.getDatasets, self._client.getDataset)
+
+    def testGetReadGroupSet(self):
+        self._testGetMethod(
+            self._repo.allReadGroupSets, self._client.getReadGroupSet)
+
+    def testGetReadGroup(self):
+        self._testGetMethod(
+            self._repo.allReadGroups, self._client.getReadGroup)
+
+    def testGetCallSet(self):
+        self._testGetMethod(
+            self._repo.allCallSets, self._client.getCallSet)
+
+    def testGetVariant(self):
+        repoVariantSets = self._repo.allVariantSets()
+        referenceName = self._repo.getReferenceSets(
+            )[0].getReferences()[0].getName()
+        for variantSet in repoVariantSets:
+            repoVariants = variantSet.getVariants(referenceName, 0, 2**32)
+            for repoVariant in repoVariants:
+                variantId = repoVariant.getId()
+                variant = self._client.getVariant(variantId)
+                self._assertEqual(repoVariant, variant)
+
+    def testGetVariantSet(self):
+        self._testGetMethod(
+            self._repo.allVariantSets, self._client.getVariantSet)
+
+    def testGetVariantAnnotationSet(self):
+        self._testGetMethod(
+            self._repo.allVariantAnnotationSets,
+            self._client.getVariantAnnotationSet)
+
+    def testGetFeatureSet(self):
+        self._testGetMethod(
+            self._repo.allFeatureSets, self._client.getFeatureSet)
+
+    def testGetFeature(self):
+        repoFeatures = self._repo.allFeatures()
+        for repoFeature in repoFeatures:
+            repoFeature = repoFeature[0]
```

Comment:
the features iterator in `sequenceAnnotations.py:Gff3DbFeatureSet.getFeatures` returns a (feature, nextPageToken) tuple ... it has the nextPageToken logic at the wrong layer of abstraction ... this is probably the same problem in the other test as well